nemo/collections/asr/parts/submodules/multitask_decoding.py

```diff
@@ -0,0 +1,493 @@
+
+    Args:
+        decoding_cfg: A dict-like object which contains the following key-value pairs.
+            strategy: str value which represents the type of decoding that can occur.
+                Possible values are :
+                -   greedy, greedy_batch (for greedy decoding).
+                -   beam, tsd, alsd (for beam search decoding).
+
+            return_scores: bool flag which determines whether to return the scores of the hypotheses.
+
+            compute_langs: a bool flag, which allows to compute language id (LID) information per token,
+                word, and the entire sample (most likely language id). The LIDS will be available
+                in the returned Hypothesis object as a dictionary
+
+            compute_hypothesis_token_set: A bool flag, which determines whether to compute a list of decoded
+                tokens as well as the decoded string. Default is False in order to avoid double decoding
+                unless required.
+
+            preserve_alignments: Bool flag which preserves the history of logprobs generated during
+                decoding (sample / batched). When set to true, the Hypothesis will contain
+                the non-null value for `alignments` in it. Here, `alignments` is a List of List of
+                Tuple(Tensor (of length V + 1), Tensor(scalar, label after argmax)).
+
+                In order to obtain this hypothesis, please utilize `rnnt_decoder_predictions_tensor` function
+                with the `return_hypotheses` flag set to True.
+
+            The config may further contain the following sub-dictionaries:
+            "greedy":
+                max_symbols: int, describing the maximum number of target tokens to decode per
+                    timestep during greedy decoding. Setting to larger values allows longer sentences
+                    to be decoded, at the cost of increased execution time.
+                preserve_frame_confidence: Same as above, overrides above value.
+                confidence_method_cfg: Same as above, overrides confidence_cfg.method_cfg.
+
+            "beam":
+                beam_size: int, defining the beam size for beam search. Must be >= 1.
+                    If beam_size == 1, will perform cached greedy search. This might be slightly different
+                    results compared to the greedy search above.
+
+                length_penalty: float, length penalty for beam search decoding. Must be >= 0.0.
+
+                max_generation_delta: int, maximum number of additional target tokens to generate
+
+                return_best_hypothesis: optional bool, whether to return just the best hypothesis or all of the
+                    hypotheses after beam search has concluded. This flag is set by default.
+
+
+        transformer_decoder: Transformer decoder module.
+        log_softmax_module: Log Softmax projection module to the vocab size.
+        tokenizer: Aggregate Tokenizer.
```

Comment:
Does it have to be an aggregate tokenizer? the typing below specifies `TokenizerSpec`. We might want to use non-aggregate tokenizer later

Reply:
Hmm good point